### PR TITLE
metadata-service[lib]: introduce a `rolloutConfiguration` field to the metadata model

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/9249

## How
* Add a `RolloutConfiguration.yaml` model
* Reference it in the connector metadata model under the `releases` field.
* Introduce a `isReleaseCandidate` field in under `releases`
* Re-generated pydantic models

### CODE REVIEW

**Comment by alafanechere** [comment]
* **#44504** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/44504?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Rollout.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://github.com/airbytehq/airbyte/airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Rollout.yaml
+title: Rollout
+description: configuration for the rollout of a connector
+type: object
+additionalProperties: false
+properties:
+  initialPercentage:
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Rollout.yaml:9]: @clnoll I was wondering if we plan to version control the rollout progress. Shall we introduce a `currentPercentage` which would be updated via automated PRs by the orchestrator?

Comment (clnoll) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Rollout.yaml:9]: @alafanechere I wasn't planning to. I'd prefer to keep the flow of information in one direction, and encourage people to use information from the RC table as the point of truth.

Comment (alafanechere) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Rollout.yaml:9]: 🆗

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Rollout.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    type: integer
+    minimum: 0
+    maximum: 100
+    default: 0
+    description: The percentage of users that should receive the new version initially.
+  maxPercentage:
+    type: integer
+    minimum: 0
+    maximum: 100
+    default: 100
```
Comment (clnoll) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Rollout.yaml:19]: WDYT about defaulting this to 50 instead? That would mean that up to half of the connectors will get the RC before it gets fully rolled out.

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorReleases.yaml [hunk 3]
```diff
@@ -8,6 +8,12 @@ additionalProperties: false
 required:
   - breakingChanges
 properties:
+  isReleaseCandidate:
```
Comment (clnoll) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorReleases.yaml:11]:
```suggestion
  isEligibleReleaseCandidate:
```

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorReleases.yaml [hunk 4]
```diff
@@ -8,6 +8,12 @@ additionalProperties: false
...
+    description: Whether the release is a release candidate.
```
Comment (clnoll) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorReleases.yaml:12]:
```suggestion
    description: Whether the release is eligible to be a release candidate.
```

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/RolloutConfiguration.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    minimum: 0
+    maximum: 100
+    default: 0
+    description: The percentage of users that should receive the new version initially.
+  maxPercentage:
+    type: integer
+    minimum: 0
+    maximum: 100
+    default: 50
+    description: The maximum percentage of users that should receive the new version.
```
Comment (clnoll) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/RolloutConfiguration.yaml:20]:
```suggestion
    description: The percentage of users that should receive the release candidate during the test phase before full rollout.
```

**Review by clnoll — APPROVED** [review]
LGTM, just a few wording suggestions.

**Comment by alafanechere** [comment]
### Merge activity

* **Aug 22, 12:07 PM EDT**: @alafanechere started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/44504).

**Pull request merged**